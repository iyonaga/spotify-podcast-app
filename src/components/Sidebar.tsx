import { signOut } from 'next-auth/react';
import Image from 'next/image';
import Link from 'next/link';
import FavoriteIcon from './Icons/Favorite';
import HomeIcon from './Icons/Home';
import PodcastIcon from './Icons/Podcast';

const menus = [
  {
    title: 'ホーム',
    href: '/',
    icon: <HomeIcon />,
  },
  {
    title: 'ポッドキャスト',
    href: '/collection/shows',
    icon: <PodcastIcon />,
  },
  {
    title: 'お気に入りのエピソード',
    href: '/collection/episodes',
    icon: <FavoriteIcon fill="white" />,
  },
];

const Sidebar = () => {
  return (
    <div className="sticky top-0 left-0 flex-none w-[250px] h-screen bg-[#2D2E39]">
      <div className="flex flex-col justify-between px-[20px] h-full">
        <ul className="mt-[30px]">
          {menus.map(({ title, href, icon }) => (
            <li key={title} className="mb-[20px]">
              <Link href={href}>
                <a className="flex items-center">
                  <span className="flex relative">{icon}</span>
                  <span className="ml-[10px] text-[16px] font-bold">
                    {title}
                  </span>
                </a>
              </Link>
            </li>
          ))}
        </ul>
        <div className="mb-[40px]">
          <button
            className="block m-auto w-[160px] h-[40px] text-[17px] font-bold text-black bg-white rounded-[30px]"
            onClick={() => signOut()}
          >
            Sign out
          </button>
        </div>
      </div>
    </div>
  );
};

export default Sidebar;
