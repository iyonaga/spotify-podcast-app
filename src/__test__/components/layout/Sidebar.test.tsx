import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { signOut } from 'next-auth/react';
import Sidebar from '@/components/layout/Sidebar';

jest.mock('next-auth/react', () => ({
  signOut: jest.fn(),
}));

jest.mock('next/router', () => ({
  useRouter() {
    return {
      pathname: '/',
    };
  },
}));

describe('components/layout/Sidebar.tsx', () => {
  test('スナップショットが一致する', () => {
    const { asFragment } = render(<Sidebar />);
    expect(asFragment()).toMatchSnapshot();
  });

  test('sign outボタンをクリックするとサインアウトが実行される', async () => {
    render(<Sidebar />);
    userEvent.click(screen.getByRole('button'));
    expect(signOut).toHaveBeenCalled();
  });
});
