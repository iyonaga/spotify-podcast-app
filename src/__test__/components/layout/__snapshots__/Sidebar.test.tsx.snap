// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/layout/Sidebar.tsx スナップショットが一致する 1`] = `
<DocumentFragment>
  <div
    class="sticky top-0 left-0 flex-none w-[250px] h-screen bg-[#2D2E39]"
  >
    <div
      class="flex flex-col justify-between px-[20px] h-full"
    >
      <ul
        class="mt-[30px]"
      >
        <li
          class="mb-[20px]"
        >
          <a
            class="flex items-center opacity-60 hover:opacity-100 transition opacity-100"
            href="/"
          >
            <span
              class="flex relative"
            >
              <svg
                fill="white"
                height="20"
                viewBox="0 0 22 20"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0.608912 6.47115C0.255472 6.68714 0.144047 7.14876 0.360039 7.5022C0.57603 7.85564 1.03765 7.96706 1.39109 7.75107L0.608912 6.47115ZM11 1L11.3911 0.360039C11.151 0.21332 10.849 0.21332 10.6089 0.360039L11 1ZM20.6089 7.75107C20.9624 7.96706 21.424 7.85564 21.64 7.5022C21.856 7.14876 21.7445 6.68714 21.3911 6.47115L20.6089 7.75107ZM19.5278 11C19.5278 10.5858 19.192 10.25 18.7778 10.25C18.3636 10.25 18.0278 10.5858 18.0278 11H19.5278ZM3.97222 11C3.97222 10.5858 3.63644 10.25 3.22222 10.25C2.80801 10.25 2.47222 10.5858 2.47222 11H3.97222ZM1.39109 7.75107L11.3911 1.63996L10.6089 0.360039L0.608912 6.47115L1.39109 7.75107ZM10.6089 1.63996L20.6089 7.75107L21.3911 6.47115L11.3911 0.360039L10.6089 1.63996ZM18.0278 11V18.1111H19.5278V11H18.0278ZM18.0278 18.1111C18.0278 18.0651 18.0651 18.0278 18.1111 18.0278V19.5278C18.8935 19.5278 19.5278 18.8935 19.5278 18.1111H18.0278ZM18.1111 18.0278H3.88889V19.5278H18.1111V18.0278ZM3.88889 18.0278C3.93489 18.0278 3.97222 18.0651 3.97222 18.1111H2.47222C2.47222 18.8936 3.10651 19.5278 3.88889 19.5278V18.0278ZM3.97222 18.1111V11H2.47222V18.1111H3.97222Z"
                  fill="white"
                />
              </svg>
            </span>
            <span
              class="ml-[10px] text-[16px] font-bold"
            >
              ホーム
            </span>
          </a>
        </li>
        <li
          class="mb-[20px]"
        >
          <a
            class="flex items-center opacity-60 hover:opacity-100 transition"
            href="/collection/shows"
          >
            <span
              class="flex relative"
            >
              <svg
                fill="white"
                height="20"
                viewBox="0 0 22 20"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M5.38636 1.90909C5.38636 1.49488 5.05058 1.15909 4.63636 1.15909C4.22215 1.15909 3.88636 1.49488 3.88636 1.90909H5.38636ZM3.88636 3.72727C3.88636 4.14149 4.22215 4.47727 4.63636 4.47727C5.05058 4.47727 5.38636 4.14149 5.38636 3.72727H3.88636ZM1.75 1C1.75 0.585786 1.41421 0.25 1 0.25C0.585786 0.25 0.25 0.585786 0.25 1H1.75ZM0.25 4.63636C0.25 5.05058 0.585786 5.38636 1 5.38636C1.41421 5.38636 1.75 5.05058 1.75 4.63636H0.25ZM18.1136 1.90909C18.1136 1.49488 17.7779 1.15909 17.3636 1.15909C16.9494 1.15909 16.6136 1.49488 16.6136 1.90909H18.1136ZM16.6136 3.72727C16.6136 4.14149 16.9494 4.47727 17.3636 4.47727C17.7779 4.47727 18.1136 4.14149 18.1136 3.72727H16.6136ZM21.75 1C21.75 0.585786 21.4142 0.25 21 0.25C20.5858 0.25 20.25 0.585786 20.25 1H21.75ZM20.25 4.63636C20.25 5.05058 20.5858 5.38636 21 5.38636C21.4142 5.38636 21.75 5.05058 21.75 4.63636H20.25ZM5.38636 8.27273C5.38636 7.85851 5.05058 7.52273 4.63636 7.52273C4.22215 7.52273 3.88636 7.85851 3.88636 8.27273H5.38636ZM18.1136 8.27273C18.1136 7.85851 17.7779 7.52273 17.3636 7.52273C16.9494 7.52273 16.6136 7.85851 16.6136 8.27273H18.1136ZM8.27273 18.4318C7.85851 18.4318 7.52273 18.7676 7.52273 19.1818C7.52273 19.596 7.85851 19.9318 8.27273 19.9318V18.4318ZM13.7273 19.9318C14.1415 19.9318 14.4773 19.596 14.4773 19.1818C14.4773 18.7676 14.1415 18.4318 13.7273 18.4318V19.9318ZM11 1.75C12.092 1.75 12.9773 2.63526 12.9773 3.72727H14.4773C14.4773 1.80683 12.9204 0.25 11 0.25V1.75ZM12.9773 3.72727V9.18182H14.4773V3.72727H12.9773ZM12.9773 9.18182C12.9773 10.2738 12.092 11.1591 11 11.1591V12.6591C12.9204 12.6591 14.4773 11.1023 14.4773 9.18182H12.9773ZM11 11.1591C9.90798 11.1591 9.02273 10.2738 9.02273 9.18182H7.52273C7.52273 11.1023 9.07956 12.6591 11 12.6591V11.1591ZM9.02273 9.18182V3.72727H7.52273V9.18182H9.02273ZM9.02273 3.72727C9.02273 2.63526 9.90798 1.75 11 1.75V0.25C9.07956 0.25 7.52273 1.80683 7.52273 3.72727H9.02273ZM3.88636 1.90909V3.72727H5.38636V1.90909H3.88636ZM0.25 1V4.63636H1.75V1H0.25ZM16.6136 1.90909V3.72727H18.1136V1.90909H16.6136ZM20.25 1V4.63636H21.75V1H20.25ZM3.88636 8.27273V9.18182H5.38636V8.27273H3.88636ZM3.88636 9.18182C3.88636 13.1106 7.07125 16.2955 11 16.2955V14.7955C7.89968 14.7955 5.38636 12.2822 5.38636 9.18182H3.88636ZM11 16.2955C14.9288 16.2955 18.1136 13.1106 18.1136 9.18182H16.6136C16.6136 12.2822 14.1003 14.7955 11 14.7955V16.2955ZM18.1136 9.18182V8.27273H16.6136V9.18182H18.1136ZM10.25 15.5455V19.1818H11.75V15.5455H10.25ZM11 18.4318H8.27273V19.9318H11V18.4318ZM11 19.9318H13.7273V18.4318H11V19.9318Z"
                />
              </svg>
            </span>
            <span
              class="ml-[10px] text-[16px] font-bold"
            >
              ポッドキャスト
            </span>
          </a>
        </li>
        <li
          class="mb-[20px]"
        >
          <a
            class="flex items-center opacity-60 hover:opacity-100 transition"
            href="/collection/episodes"
          >
            <span
              class="flex relative"
            >
              <svg
                fill="white"
                height="20"
                viewBox="0 0 22 20"
                width="22"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M21 6.86222C21 8.4087 20.4062 9.8941 19.3458 10.9929C16.9049 13.523 14.5374 16.1613 12.0053 18.5997C11.4249 19.1505 10.5042 19.1304 9.9488 18.5547L2.65376 10.9929C0.448748 8.7072 0.448748 5.01723 2.65376 2.73157C4.88044 0.42345 8.50794 0.42345 10.7346 2.73157L10.9998 3.00642L11.2648 2.73173C12.3324 1.6245 13.7864 1 15.3053 1C16.8242 1 18.2781 1.62444 19.3458 2.73157C20.4063 3.83045 21 5.31577 21 6.86222Z"
                  stroke="white"
                  stroke-linejoin="round"
                  stroke-width="1.5"
                />
              </svg>
            </span>
            <span
              class="ml-[10px] text-[16px] font-bold"
            >
              お気に入りのエピソード
            </span>
          </a>
        </li>
      </ul>
      <div
        class="mb-[40px]"
      >
        <button
          class="block m-auto w-[160px] h-[40px] text-[17px] font-bold text-black bg-white hover:bg-white/80 rounded-[30px] transition duration-500"
        >
          Sign out
        </button>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
